pkg/cluster/logs/logs.go

```diff
@@ -65,8 +68,24 @@ func Collect(nodes []nodes.Node, dir string) error {
 	// add a log collection method for each node
 	for _, n := range nodes {
 		node := n // https://golang.org/doc/faq#closures_and_goroutines
+		name := node.String()
+		// grab all logs under /var/log (pods and containers)
+		cmd := osexec.Command("docker", "exec", node.Name(), "tar", "--hard-dereference", "-C", "/var/log", "-chf", "-", ".")
```

Comment:
why aren't we using the container cmder here?

Reply:
I would prefer to isolate which code execs to docker, we're going to shim that soon and it will be much easier to fix if it is in less places.

Reply:
There are 2 reasons because I couldn´t use the container cmder:
* it doesn´t have implemented .StdoutPipe()
* it executes docker with the `-t` that allocates a tty and  makes the `tar` command to fail

Do you see an alternative to implementing StdoutPipe() ?

> I would prefer to isolate which code execs to docker, we're going to shim that soon and it will be much easier to fix if it is in less places.

You mean docker insider the nodes or docker everywhere?

Reply:
> /ok-to-test
> you should apply to join kubernetes-sigs :-)

Sure, that will be great, do you know if is this process the one I have to follow? https://github.com/kubernetes/community/blob/master/community-membership.md

Do I need 2 sponsors?

Reply:
> Do you see an alternative to implementing StdoutPipe() ?

stdout should not be a problem to pipe, we collect stdout regularly.

> it executes docker with the -t that allocates a tty and makes the tar command to fail

the actual code does:
```go
	if c.stderr != nil || c.stdout != nil {
		args = append(args,
			"-t", // use a tty so we can get output
		)
	}
```

we can make this code handle our case

Reply:
> You mean docker insider the nodes or docker everywhere?

everywhere. we need to be able to use a shim eventually so the code is testable and can support podman properly.

using the node cmder will also make it easier to fake out the node.

Reply:
in what way does tar fail with a TTY?

Reply:
I got this working pretty nicely with the node cmder, breaking it into multiple PRs, first https://github.com/kubernetes-sigs/kind/pull/473

Reply:
see: https://github.com/kubernetes-sigs/kind/compare/master...BenTheElder:pr458?expand=1

https://github.com/kubernetes-sigs/kind/pull/474 adds the other required peice